src/vis/violin/ViolinSegmentedControl.tsx

```diff
@@ -0,0 +1,47 @@
+import { Input, SegmentedControl, Tooltip } from '@mantine/core';
+import * as React from 'react';
+import { EViolinSeparationMode, EViolinOverlay } from './interfaces';
+
+interface SegmentedControlProps<T> {
+  callback: (s: T) => void;
+  currentSelected: T;
+  disabled?: boolean;
+}
+
+export function ViolinOverlaySegmentedControl({ callback, currentSelected, disabled }: SegmentedControlProps<EViolinOverlay>) {
+  return (
+    <Input.Wrapper label="Overlay">
+      <SegmentedControl
+        disabled={disabled}
+        fullWidth
+        size="xs"
+        value={currentSelected}
+        onChange={callback}
+        data={[
+          { label: EViolinOverlay.NONE, value: EViolinOverlay.NONE },
+          { label: EViolinOverlay.BOX, value: EViolinOverlay.BOX },
+        ]}
+      />
+    </Input.Wrapper>
+  );
+}
+
+export function ViolinSeparationSegmentedControl({ callback, currentSelected, disabled }: SegmentedControlProps<EViolinSeparationMode>) {
+  return (
+    <Input.Wrapper label="Separation">
+      <Tooltip label={disabled ? 'Faceting not possible' : 'Group within plot or split into facets'} withArrow>
```

Comment:
![grafik](https://github.com/datavisyn/visyn_core/assets/5851088/da9b625d-f100-4c81-a0b6-5417de080a6f)

Can we extend/change the tooltip in a way that it explains what users must change to enable this feature (e.g., remove categorical column select another numerical column)? 

```suggestion
      <Tooltip label={disabled ? 'The separation is disabled. To enable, deselect the categorical column or add another numerical column above.' : 'Group within plot or split into facets'} withArrow>
```

Feel free to adapt my tooltip suggestion.

cc @dvchristianbors

Reply:
@dvchristianbors  and me fixed the logic and the tooltip text together in a call.